ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorGroupByOperator.java

```diff
@@ -514,7 +526,8 @@ private void prepareBatchAggregationBufferSets(VectorizedRowBatch batch) throws
           // is very important to clone the keywrapper, the one we have from our
           // keyWrappersBatch is going to be reset/reused on next batch.
           aggregationBuffer = allocateAggregationBuffer();
```

Comment:
allocateAggregationBuffer can be reused with the flushed out entries.

Reply:
@rbalamohan I have another patch for that.